ibc-core/ics24-host/types/src/error.rs

```diff
@@ -9,6 +9,35 @@ use ibc_primitives::prelude::*;
 use ibc_primitives::proto::Error as ProtoError;
 use prost::DecodeError as ProstError;
 
+/// Errors that originate from host implementations.
+#[derive(Debug, Display)]
+pub enum HostError {
+    /// application module error: `{description}`
+    AppModule { description: String },
```

Comment:
We don't expect any application specific errors be stemming from hosts. As applications are on top of the IBC core stack. I beleive, this variant should later be part of the ICS-26 router errors, once we refactored the `Router` trait. I suggest we pick the `Other` for places the `AppModule` is already used.

Reply:
[2a53a7d](https://github.com/cosmos/ibc-rs/pull/1331/commits/2a53a7dbf5e4c934c56ef998d0d77d1e5bc9c204)